### PR TITLE
[ROCm][CI] add warmup to mem_util test before measurement

### PR DESCRIPTION
**Pull request by divakar-amd**
This PR adds a warmup step to the CUDA/ROCm context in the `test_memory_profiling` test to ensure that PyTorch's internal initialization overhead is included in the memory baseline. This change helps to produce more accurate memory profiling results, especially on ROCm where the initialization overhead can be larger.

This PR ensures PyTorch's initialization overhead is included in memory measurements.

This PR resolves the following CI test for ROCm.
`pytest -s -v utils_/test_mem_utils.py::test_memory_profiling`

CI run reference: https://buildkite.com/vllm/amd-ci/builds/8525/list?sid=019e2ade-0a7a-49e4-8c9a-b1ba4355e10f&tab=output

### CODE REVIEW

File: tests/utils_/test_mem_utils.py [hunk 1]
```diff
@@ -20,6 +20,13 @@ def test_memory_profiling():
     # 512 MiB allocation outside of this instance
     handle1 = lib.cudaMalloc(512 * 1024 * 1024)
 
+    # Warm up PyTorch's CUDA/ROCm context so that its internal initialization
+    # overhead (streams, cuBLAS handles, etc.) is included in the baseline and
+    # does not inflate non-torch increase which is larger on ROCm than on CUDA
+    _warmup = torch.zeros(1, device="cuda")
```
Comment (tjtanaa) [inline tests/utils_/test_mem_utils.py:26]: so cuda also needs this?

Comment (divakar-amd) [reply tests/utils_/test_mem_utils.py:26]: The same overhead is also present on Cuda, yes. However, it is not as big and luckily passes the 5% tolerance set in the test [here](https://github.com/ROCm/vllm/blob/03993b63f541b5b8ef15d5c8d09eaf15b5ce08e1/tests/utils_/test_mem_utils.py#L64)

**Review by zou3519 — APPROVED** [review]
